### ISSUE TITLE
Windows 10 install fails with "cannot open include file dns_sd.h"

### ISSUE THREAD

**Issue by hageson** [nfarina/homebridge#795]
I followed the Wiki instructions but the homebridge install keeps failing.  The first errors reported are that the include file dns_sd.h could not be found.  I have searched for similar issues to no avail and have tried everything I can think of but I am new to this so my knowledge is rudimentary, to say the least.  Hopefully someone can see what the problem is?

I have a 64bit Windows 10 machine with all the latest updates.  I use BitDefender as my antivirus.  My C drive is relatively small so when I install anything that gives me an option to change the install location, I change it to the D drive.  For example, Visual Studio.  Could that be causing a problem?

From the cmd console:

C:\Users\Howard\AppData\Roaming\npm\homebridge -> C:\Users\Howard\AppData\Roaming\npm\node_modules\homebridge\bin\homebridge

> mdns@2.3.3 install C:\Users\Howard\AppData\Roaming\npm\node_modules\homebridge\node_modules\mdns
> node-gyp rebuild

C:\Users\Howard\AppData\Roaming\npm\node_modules\homebridge\node_modules\mdns>if not defined npm_config_node_gyp (node "C:\Users\Howard\AppData\Roaming\npm\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (node "" rebuild )
Building the projects in this solution one at a time. To enable parallel build, please add the "/m" switch.
  dns_sd.cpp
  dns_service_browse.cpp
  dns_service_enumerate_domains.cpp
  dns_service_get_addr_info.cpp
  dns_service_process_result.cpp
  dns_service_ref.cpp
  dns_service_ref_deallocate.cpp
  dns_service_ref_sock_fd.cpp
c:\users\howard\appdata\roaming\npm\node_modules\homebridge\node_modules\mdns\src\mdns.hpp(32): fatal error C1083: Cannot open include file: 'dns_sd.h': No such file or directory (compiling source file ..\src\dns_sd.cpp) [C:\Users\Howard\AppData\Roaming\npm\node_modules\homebridge\node_modules\mdns\bui
ld\dns_sd_bindings.vcxproj]

This error is repeated a number of times.  Here is a full transcript of the messages displayed:

https://1drv.ms/w/s!AnK_dzLRdrSKxFTH9UY5jGYK7-PQ

The complete debug log is attached:

[npm-debug.zip](https://github.com/nfarina/homebridge/files/490799/npm-debug.zip)

**Comment by lino-silva** [comment]
Have you got Bonjour installed on your machine? You might need to download and install it. It's bundled with Apple's `Bonjour Print Services for Windows`

[Bonjour Print Services for Windows](https://support.apple.com/kb/DL999?locale=en_US)

**Comment by hageson** [comment]
Yes, I have installed Bonjour.  I can see the file -> C:\Program Files\Bonjour SDK\Include\dns_sd.h  But is the install script looking for it there?

**Comment by hageson** [comment]
Just in case I confused the issue, I closed this issue by mistake.  It is still open.

**Comment by KhaosT** [comment]
@hageson can you check and ensure ``BONJOUR_SDK_HOME`` is set? [Via node_mdns](https://github.com/agnat/node_mdns/blob/master/README.md)

**Comment by hageson** [comment]
The BONJOUR_SDK_HOME environmental system variable was set to "D:\Program Files\Bonjour SDK\".  I changed it to C:\Program Files\Bonjour SDK and the dns_sd.h error messages are now gone!  Hooray!  

Unfortunately, I have a raft of other warnings and error messages now.  But this particular error is resolved so I will close this issue.  Thanks to lino-silva and KhaosT for pointing me in the right direction.

For the record, as I said earlier, given the option, I always change the install location of anything I install to drive D.  Which is what I did when I installed the Bonjour SDK.  Turns out that the only component of the SDK that was installed on the D drive was the SDK documentation "About Bonjour SDK.rtf".  The actual SDK was installed on the C drive.
